File: app/src/main/java/workshop/akbolatss/xchangesrates/repositories/DBOpenHelper.java
package workshop.akbolatss.xchangesrates.repositories;

import android.content.Context;

import org.greenrobot.greendao.database.Database;

import workshop.akbolatss.xchangesrates.model.dao.DaoMaster;

/**
 * Author: Akbolat Sadvakassov
 * Date: 25.01.2018
 */

public class DBOpenHelper extends DaoMaster.OpenHelper {

    public DBOpenHelper(Context context, String name) {
        super(context, name);
    }

    @Override
    public void onUpgrade(Database db, int oldVersion, int newVersion) {
        super.onUpgrade(db, oldVersion, newVersion);

        switch (oldVersion){
            case 1://version one
                break;
            case 2://version two
                break;
        }
    }
}

File: app/src/main/java/workshop/akbolatss/xchangesrates/repositories/DBOpenHelper.kt
package workshop.akbolatss.xchangesrates.repositories

import android.content.Context

import org.greenrobot.greendao.database.Database

import workshop.akbolatss.xchangesrates.model.dao.DaoMaster

/**
 * Author: Akbolat Sadvakassov
 * Date: 25.01.2018
 */

class DBOpenHelper(context: Context, name: String) : DaoMaster.OpenHelper(context, name) {

    override fun onUpgrade(db: Database?, oldVersion: Int, newVersion: Int) {
        super.onUpgrade(db, oldVersion, newVersion)

        when (oldVersion) {
            1//version one
            -> {
            }
            2//version two
            -> {
            }
        }
    }
}
